docs/migration-guides.md

```diff
@@ -0,0 +1,1356 @@
+
+  return [assetIdx, appIdx, accIdx] as const
+}
+```
+
+Alternatively, update the implementation to use default `value` resource encoding.
+
+```ts
+// ... rest of code
+
+test(asset: Asset, app: Application, acc: Account): [Asset, Application, Account] {
+  const assetId = op.btoi(Txn.applicationArgs(1))
+  assert(asset === Asset(assetId), 'expected asset to be passed by value')
+
+  const appId = op.btoi(Txn.applicationArgs(2))
+  assert(app === Application(appId), 'expected application to be passed by value')
+
+  const address = Txn.applicationArgs(3)
+  assert(acc === Account(address), 'expected account to be passed by value')
+
+  return [asset, app, acc]
+}
+```
+
+#### Rename test files from `.(spec|test).ts` to `.algo.(spec|test).ts`.
+
+Existing test files that need to run in a simulated AVM environment or import modules from `algorand-typescript` or `algorand-typescript-testing` packages should be renamed to have either the `.algo.spec.ts` or `.algo.test.ts` extension.
+
+`puyaTsTransformer` will only provide the simulated AVM environment and executable implementations of `algorand-typescript` module to files with those extensions, excluding standard `.spec.ts` or `.test.ts` files (e.g. `e2e.spec.ts`).
+
+This change can improve test performance by ensuring only relevant files are processed through `algorandTsTransformer`.
+
+#### Rename `arc4EncodedLength` function to `sizeOf`
+
+**BEFORE - Algorand TypeScript beta**
+
+```ts
+import { arc4EncodedLength, assert } from '@algorandfoundation/algorand-typescript'
+
+// ... rest of code
+
+assert(arc4EncodedLength<uint64>() === 8);
+assert(arc4EncodedLength<boolean>() === 1);
+assert(arc4EncodedLength<UintN<512>>() === 64);
+assert(arc4EncodedLength<[StaticArray<Bool, 10>, boolean, boolean]>() === 3);
+```
+
+**AFTER - Algorand TypeScript 1.0**
+
+```ts
+import { sizeOf, assert } from '@algorandfoundation/algorand-typescript'
+
+// ... rest of code
+
+assert(sizeOf<uint64>() === 8);
+assert(sizeOf<boolean>() === 1);
+assert(sizeOf<Uint<512>>() === 64);
+assert(sizeOf<[StaticArray<Bool, 10>, boolean, boolean]>() === 3);
+```
+
+#### Update `abiCall` syntax to use object parameter
+
+The signature of `abiCall` helper changes from
+
+```ts
+import { arc4 } from '@algorandfoundation/algorand-typescript'
+import MyContract from './MyContract.algo'
+
+// ... rest of code
+
+arc4.abiCall(MyContract.prototype.myMethod, { ... })
+```
+
+to
+
+```ts
+import { arc4 } from '@algorandfoundation/algorand-typescript'
+import MyContract from './MyContract.algo'
+
+// ... rest of code
+
+arc4.abiCall({ method: MyContract.prototype.myMethod, ... })
+```
+
+The new `method` property exists to provide a natural way to specify the `TMethod` generic parameter, but it is optional and alternatively the generic arg can be specified explicitly with
+
+```ts
+import { arc4 } from '@algorandfoundation/algorand-typescript'
+import MyContract from './MyContract.algo'
```

Comment:
nitpick: it could be changed to `import type MyContract from './MyContract.algo'` to demonstrate type only import works with this approach.

Reply:
done